clang/test/CXX/drs/cwg2398.cpp

```diff
@@ -0,0 +1,115 @@
+// RUN: %clang_cc1 %s -fsyntax-only -std=c++23                                     -verify=expected,new
+// RUN: %clang_cc1 %s -fsyntax-only -std=c++23 -fno-relaxed-template-template-args -verify=expected,old
```

Comment:
We don't test non-conforming modes of operation in DR test suite.
This should be tested elsewhere, e.g. in SemaTemplate tests.

Reply:
Okay, I think I misunderstood then what `CXX/drs` was about, I should probably move it elsewhere.

Reply:
Given that you're fixing CWG2398, you should be adding a test here to update https://clang.llvm.org/cxx_dr_status.html. This corner of our test suite lives by its own rules to some extent, so you should stay consistent with 700 tests we already have.

Reply:
Thanks.

Since the core issue is not resolved by WG21, I don't want to claim to be resolving it here with this patch.

And while we don't remove the non-conforming flag, we should still test it, and it's just much more economical and error proof that we reuse the same tests.

Reply:
> Since the core issue is not resolved by WG21, I don't want to claim to be resolving it here with this patch.

It'd be far from the first time we're testing an issue that is still open (https://github.com/llvm/llvm-project/blob/bd53c7cce418fe7f3e171859d4718df15d03dc2b/clang/test/CXX/drs/dr24xx.cpp#L62), but the fact that 2398 filing lacks any possible resolution that we can be testing against, it complicates the matter.